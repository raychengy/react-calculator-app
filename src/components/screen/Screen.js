import React from 'react';
import PropTypes from 'prop-types';
import ResultScreen from './ResultScreen';
import ComputationScreen from './ComputationScreen';

const Screen = ({ equation, results }) => (
    <div className="screen">
        <ResultScreen value={results} />
        <ComputationScreen value={equation} />
    </div>
)

Screen.propTypes = {
    results: PropTypes.string.isRequired,
    equation: PropTypes.string.isRequired,
}

export default Screen;