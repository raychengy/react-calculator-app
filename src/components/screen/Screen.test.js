import React from 'react';
import renderer from 'react-test-renderer';

import Screen from './Screen';

const equation = '';
const results = "0";


describe('<Screen />', () => {
    describe('Using props', () => {
        it('renders without crashing', () => {
            const screen = renderer.create(<Screen
                equation={equation}
                results={results}
            />);
            expect(screen.toJSON()).toMatchSnapshot();
        });
    })
})