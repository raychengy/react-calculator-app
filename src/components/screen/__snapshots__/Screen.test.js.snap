// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="screen"
>
  <div
    className="result-screen"
  >
    0
  </div>
  <div
    className="computation-screen"
  >
    
  </div>
</div>
`;
